src/server/db_slice.cc

```diff
@@ -1193,6 +1195,36 @@ int32_t DbSlice::GetNextSegmentForEviction(int32_t segment_id, DbIndex db_ind) c
          db_arr_[db_ind]->prime.GetSegmentCount();
 }
 
+void DbSlice::ScheduleForOffloadStep(DbIndex db_indx, size_t increase_goal_bytes) {
+  VLOG(1) << "ScheduleForOffloadStep increase_goal_bytes:"
+          << strings::HumanReadableNumBytes(increase_goal_bytes);
+  DCHECK(shard_owner()->tiered_storage());
+  FiberAtomicGuard guard;
+  PrimeTable& pt = db_arr_[db_indx]->prime;
+
+  static PrimeTable::Cursor cursor;
+
+  size_t offloaded_bytes = 0;
+  auto cb = [&](PrimeIterator it) {
+    // TBD check we did not lock it for future transaction
+
+    if (increase_goal_bytes > offloaded_bytes && !(it->first.HasTouched()) &&
+        TieredStorage::CanExternalizeEntry(it)) {
+      shard_owner()->tiered_storage()->ScheduleOffload(db_indx, it);
+      if (it->second.HasIoPending()) {
+        offloaded_bytes += it->second.Size();
+        VLOG(2) << "ScheduleOffload bytes:" << offloaded_bytes;
+      }
+    }
+    it->first.SetTouched(false);
```

Comment:
you only need to do so if the slot has been touched and will be offloaded.. so it should go under the if-condition.

Reply:
We offload only items that are marked as not touched, which means that they are cold items.
How do we make them cold (set as not touched), by iterating on the database and setting touched to false. If the item was fetched it is marked touched true. This way if the item was no touched from the last time we iterated on this bucket it means that it is cold and a candidate for offload.
Therefor the SetTouched(false) should be outside the if